cloud-sql/mysql/Startup.cs

```diff
@@ -45,48 +43,105 @@ public Startup(IConfiguration configuration)
-            services.AddScoped(typeof(DbConnection), (IServiceProvider) =>
-                InitializeDatabase());
+            services.AddSingleton(sp => new MySqlConnection().GetMySqlConnectionString());
             services.AddMvc(options =>
             {
                 options.Filters.Add(typeof(DbExceptionFilterAttribute));
             });
-            _services = services;
         }
 
-        DbConnection InitializeDatabase()
+        // This method gets called by the runtime. Use this method to configure the HTTP request pipeline.
+        public void Configure(IApplicationBuilder app, IHostingEnvironment env,
+            ILoggerFactory loggerFactory)
         {
-            DbConnection connection;
+            loggerFactory.AddConsole(Configuration.GetSection("Logging"));
+
+            if (env.IsDevelopment())
+            {
+                app.UseDeveloperExceptionPage();
+            }
+            else
+            {
+                // Configure error reporting service.
+                app.UseGoogleExceptionLogging();
+                app.UseExceptionHandler("/Home/Error");
+            }
+
+            app.UseMvc(routes =>
+            {
+                routes.MapRoute(
+                    name: "default",
+                    template: "{controller=Home}/{action=Index}/{id?}");
+            });
+        }
+    }
+
+    static class StartupExtensions
+    {
+        public static DbConnection OpenWithRetry(this DbConnection connection) =>
+            // [START cloud_sql_mysql_dotnet_ado_backoff]
+            Policy
+                .HandleResult<DbConnection>(conn => conn.State != ConnectionState.Open)
+                .WaitAndRetry(new[]
+                {
+                    TimeSpan.FromSeconds(1),
+                    TimeSpan.FromSeconds(2),
+                    TimeSpan.FromSeconds(5)
+                }, (result, timeSpan, retryCount, context) =>
+                {
+                    // Log any warnings here.
+                    if(result.Exception != null && retryCount == 3)
+                    {
+                        throw result.Exception;
+                    }
+                })
+                .Execute(() =>
+                {
+                    // Return a new connection.
+                    try {
+                        connection.Open();
+                    }
+                    catch (MySqlException e)
+                    {
+                         Console.WriteLine(
+                             $"Error connecting to database: {e.Message}");
+                    }
+                    return connection;
+                });
+            // [END cloud_sql_mysql_dotnet_ado_backoff]
 
-            connection = GetMySqlConnection();
-            connection.Open();
-            using (var createTableCommand = connection.CreateCommand())
+        public static void InitializeDatabase(this DbConnection connection)
+        {
+            using(connection.OpenWithRetry())
             {

```

Comment:
Because of the proposed changes in Polly, now something like this needs to be done:
```suggestion
        public static void InitializeDatabase()
        {
            var connectionString = GetMySqlConnectionString();
            using(DbConnection connection = new MySqlConnection(connectionString.ConnectionString))
            {
                connection.OpenWithRetry();
```

Reply:
Done.